src/daemon/http/mod.rs

```diff
@@ -433,67 +452,23 @@ impl Request {
         self.read_bytes(limit).await
     }
 
-    /// See hyper::body::to_bytes
-    ///
-    /// Here we want to limit the bytes consumed to a maximum. So, the
-    /// code below is adapted from the method in the hyper crate.
     pub async fn read_bytes(self, limit: u64) -> Result<Bytes, Error> {
-        let body = self.request.into_body();
-
-        futures_util::pin_mut!(body);
+        // We’re going to cheat a bit. If we know the body is too big from
```

Comment:
It's not clear to me if this code, or even the old code, actually prevents reading too much data into memory from the network.

Reply:
My understanding is that Hyper will only start reading the body when you explicitly tell it to. The first test here basically looks at the Content-Length header and rejects the request if that’s too large before even beginning to read the body. The new `Limited` thing is only relevant for chunked transfers.